README.md

```diff
@@ -1,45 +1,62 @@
+## Api config
+To test the api locally fill the variable `BASE_API_ENDPOINT_URL` with `http://localhost:3000/api` and run the api locally, you can follow the tutorial to set up the api locally. To test the api on development, just fill `BASE_API_ENDPOINT_URL` with `https://devapi.collaction.org`.
 
-Make a copy of the `.env.example` file, name it `.env` and fill out the missing secrets.
+## Install the dependencies
+On the root path of the project run `flutter pub get` on the command.
 
-## Firebase
+## Firebase configuration
+Firebase config is managed by `FirebaseOptionsFactory` inside `generate_firebase_options.dart` this factory is going to generate a `FirebaseOptions` object based on `env` values. So no action is needed.
 
-TBD
+## Generating Code
+In order to generate code related to `freezed` , `injectable` , `json_serializable`. Choose a command to run when you need to build or re-build these files.
 
-## Getting Started
+The watch command keeps re-building generated files. It’s generally not recommended to use
 
-This project is a starting point for a Flutter application.
+ `flutter pub run build_runner watch --delete-conflicting-outputs`
 
-A few resources to get you started if this is your first Flutter project:
+If you want to build generated files once run this command, It’s preferred.
 
-- [Lab: Write your first Flutter app](https://flutter.dev/docs/get-started/codelab)
-- [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)
+`flutter pub run build_runner build --delete-conflicting-outputs`
 
-For help getting started with Flutter, view our
-[online documentation](https://flutter.dev/docs), which offers tutorials,
-samples, guidance on mobile development, and a full API reference.
+Sometimes, the values of the generated files don’t update properly, If that’s the case, one solution could be to run the following command before running the above commands.
+
+`flutter pub run build_runner clean`.
+
+## Running the project
+
+**Firebase Emulator (Preferred method)**
+
+After running `build_runner` command, run `flutter build web` command to build the web directory. To test the routes it’s important to create a `404.html` file inside web/build and copy the content of the file `index.html` into `404.html` . This way of running the project should work with the **api** running locally, as long as the `ADMINCMS_URL` variable in `.env` file of the **api** project is set to `http://localhost:55714`.
+
+The command to run the firebase emulator is: `firebase emulators:start`.
+
+**Flutter run**
+
+Run the command  `flutter run -d chrome --web-port=<ADMINCMS_URL port>`.
+
+:warning: **Considerations**
+
+It has been reported some memory problems because flutter appears to generate files in `.../Local/AppData/Temp` folder when running on chrome `flutter run -d chrome` . So from time to time is recommended to check that folder if there is presence of `flutter_tools` folders and erase them. 
```

Comment:
```suggestion
It has been reported some memory problems because Flutter appears to generate files in `.../Local/AppData/Temp` folder when running on Chrome `flutter run -d chrome` . Therefore it is recommended to check that folder for the presence of `flutter_tools` folders and erase them. 
```